controllers/category.js

```diff
@@ -201,21 +203,46 @@ exports.getOne = async (req, res) => {
  *       "error": categoryNameNotFound"
  *     }
  */
+
 exports.getAll = async (req, res) => {
   try {
-    const { page, perpage, categoryName } = req.query;
-    const options = {
-      page: parseInt(page, 10) || 1,
-      limit: parseInt(perpage, 10) || 10,
-      sort: { categoryName: 1 },
-    };
-    const category = await Category.paginate(
-      { categoryName: { $regex: new RegExp(categoryName), $options: "i" } },
-      options
-    );
-    res.status(200).json({ category });
-  } catch (err) {
-    res.status(500).json(err);
+    const { limit = 10, page = 1, categoryName, subCategories } = req.query;

```

Comment:
why we are taking two different variable category name and sub category. We should have only one search field in frontend